### PR TITLE
TYP: some types for pandas/io/formats/csvs.py

### PR DESCRIPTION
**Pull request by simonjayhawkins**
broken off #28339

### CODE REVIEW

File: pandas/io/formats/csvs.py [hunk 1]
```diff
@@ -187,7 +188,7 @@ def save(self):
             close = True
 
         try:
-            writer_kwargs = dict(
+            kwargs: Dict[str, Any] = dict(
```
Comment (jbrockmendel) [inline pandas/io/formats/csvs.py]: why the rename?  ive recently been finding more-specific names for kwargs useful when tracking down less-specific usages

Comment (simonjayhawkins) [reply pandas/io/formats/csvs.py]: no reason other than for consistency of variable names. will revert tomorrow.

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by jbrockmendel** [comment]
LGTM; CI failures are unrelated

**Comment by simonjayhawkins** [comment]
rebased + green

**Comment by jreback** [comment]
thanks @simonjayhawkins

**Pull request merged**